internal/task/task_test.go

```diff
@@ -155,11 +187,15 @@ func TestTask_GetCPU_GetsCPUUsingContainerLimit(t *testing.T) {
 			ts := tt.testServer(tt.containerCPU, tt.taskCPU)
 			defer ts.Close()
 
-			ecsTask, err := task.New(config.Config{MetadataURI: ts.URL, ConainerID: "container-id"})
+			ecsTask, err := task.New(config.Config{MetadataURI: ts.URL})
 			assert.NoError(t, err)
 
 			gotCPU, err := ecsTask.GetMaxProcs()
-			assert.NoError(t, err)
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
```

Comment:
are we able to move the error case into a seperate test? Only reason being is because then it avoids any conditional logic in our tests and our tests then focus on testing one specific thing. It will also allow us to add extra assertions on the error like the expected error message, keep our tests relatively simple and enable us to extend the error test in the future in case any extra error cases arise. Hope thats ok.